src/System.CommandLine.Subsystems/ErrorReportingSubsystem.cs

```diff
@@ -1,24 +1,49 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.CommandLine.Parsing;
 using System.CommandLine.Subsystems;
 using System.CommandLine.Subsystems.Annotations;
 
 namespace System.CommandLine;
 
+/// <summary>
+/// Subsystem for reporting errors
+/// </summary>
+/// <remarks>
+/// This class, including interface, is likey to change as powderhouse continues
+/// </remarks>
 public class ErrorReportingSubsystem : CliSubsystem
 {
     public ErrorReportingSubsystem(IAnnotationProvider? annotationProvider = null)
         : base(ErrorReportingAnnotations.Prefix, SubsystemKind.ErrorReporting, annotationProvider)
     { }
 
-    // TODO: Stash option rather than using string
     protected internal override bool GetIsActivated(ParseResult? parseResult)
         => parseResult is not null && parseResult.Errors.Any();
 
+    // TODO: properly test execute directly when parse result is usable in tests
     protected internal override CliExit Execute(PipelineContext pipelineContext)
     {
-        pipelineContext.ConsoleHack.WriteLine("You have errors!");
+        var _ = pipelineContext.ParseResult
+            ?? throw new ArgumentNullException($"{nameof(pipelineContext)}.ParseResult");
+
+        Report(pipelineContext.ConsoleHack, pipelineContext.ParseResult.Errors);
+
         return CliExit.SuccessfullyHandled(pipelineContext.ParseResult);
     }
+
+    public void Report(ConsoleHack consoleHack, IReadOnlyList<ParseError> errors)
+    {
+        ConsoleHelpers.ResetTerminalForegroundColor();
+        ConsoleHelpers.SetTerminalForegroundRed();
```

Comment:
Is a Reset needed if we are about to set it?
```suggestion
        ConsoleHelpers.SetTerminalForegroundRed();
```

Reply:
This code is just pulled from the existing `WriteErrorDetails` method on `ParseErrorAction` from main.

https://github.com/dotnet/command-line-api/blob/963d34b1fb712c673bfb198133d7e988182c9ef4/src/System.CommandLine/Invocation/ParseErrorAction.cs#L47-L60

The idea is pulling in that existing code from `main` into `main-powderhouse` we return `main` functionality over `main-powderhouse`'s current `pipelineContext.ConsoleHack.WriteLine("You have errors!");`. I think it's quite likely that this needs to change / be updated but I'm not sure that should be done in this PR

Reply:
I reference this in my other comment, but `parseResult.Configuration.Error.WriteLine();` on line 54 is why I added a zero-arity version of `WriteLine` in this PR